vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/features/metadata/source/exif/ExifMetadataExtractor.kt

```diff
@@ -0,0 +1,63 @@
+package org.vitrivr.engine.core.features.metadata.source.exif
+
+import com.drew.imaging.ImageMetadataReader
+import com.drew.metadata.Directory
+import com.drew.metadata.Metadata
+import org.vitrivr.engine.core.features.AbstractExtractor
+import org.vitrivr.engine.core.model.content.element.ContentElement
+import org.vitrivr.engine.core.model.descriptor.struct.metadata.source.MapStructDescriptor
+import org.vitrivr.engine.core.model.metamodel.Schema
+import org.vitrivr.engine.core.model.retrievable.Retrievable
+import org.vitrivr.engine.core.model.retrievable.attributes.SourceAttribute
+import org.vitrivr.engine.core.operators.Operator
+import org.vitrivr.engine.core.source.MediaType
+import org.vitrivr.engine.core.source.file.FileSource
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+
+class ExifMetadataExtractor(
+    input: Operator<Retrievable>,
+    field: Schema.Field<ContentElement<*>, MapStructDescriptor>,
+    persisting: Boolean = true
+) : AbstractExtractor<ContentElement<*>, MapStructDescriptor>(input, field, persisting, bufferSize = 1) {
+    override fun matches(retrievable: Retrievable): Boolean =
+        retrievable.filteredAttribute(SourceAttribute::class.java)?.source is FileSource /* && retrievable.type is MediaType.IMAGE */
+
+
+    fun convertType(directory: Directory, tagType: Int, type: String): Any? {
```

Comment:
I do not see why the type is a String, when there is a whole type system implemented in vitrivr-engine?